linode/framework_provider.go

```diff
@@ -0,0 +1,111 @@
+package linode
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/provider"
+	"github.com/hashicorp/terraform-plugin-framework/provider/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/linode/terraform-provider-linode/linode/helper"
+	"github.com/linode/terraform-provider-linode/linode/token"
+)
+
+type FrameworkProvider struct {
+	ProviderVersion string
+	Meta            *helper.FrameworkProviderMeta
+}
+
+func CreateFrameworkProvider(version string) provider.Provider {
+	return &FrameworkProvider{
+		ProviderVersion: version,
+	}
+}
+
+func (p *FrameworkProvider) Metadata(
+	ctx context.Context,
+	req provider.MetadataRequest,
+	resp *provider.MetadataResponse,
+) {
+	resp.TypeName = "Linode cloud"
```

Comment:
Judging by the framework documentation, it looks like we might just want this string to become something like `linode`: https://developer.hashicorp.com/terraform/plugin/framework/resources#metadata-method

Reply:
Good catch, thank you!